src/inverse_cai/data/annotated_pairs_format.py

```diff
@@ -40,21 +40,28 @@ def hash_comparison(text_a: str, text_b: str, prompt: Optional[str]) -> str:
 
 
 def votes_to_annotations(
-    votes: Mapping[int, Optional[bool]],
+    votes: Mapping[int, Union[bool, str, None]],
     principle_index_to_text: Mapping[int, str],
     active_principles: Sequence[str],
     reference_preference: str,
-) -> Dict[str, Dict[str, str]]:
+) -> Dict[str, Dict[str, Optional[str]]]:
     """Convert principle votes to annotations in the standardized format.
 
     Args:
-        votes: Dictionary mapping principle IDs to their votes (True/False/None)
+        votes: Dictionary mapping principle IDs to their votes:
+            - True: Principle agrees with reference preference
+            - False: Principle disagrees with reference preference
+            - None: Principle not applicable to this comparison
+            - "invalid": Vote is invalid for technical reasons
         principle_index_to_text: Dictionary mapping principle IDs to their text
         active_principles: List of active principles to include
         reference_preference: The reference preference (text_a or text_b)
 
     Returns:
-        Dictionary mapping principle IDs (hashed) to dictionaries with preferences
+        Dictionary mapping principle IDs (hashed) to dictionaries with preference information:
+        - For True/False votes: {"pref": "text_a" or "text_b"}
+        - For None votes: {"pref": None, "no_pref_reason": "not_applicable"}
+        - For "invalid" votes: {"pref": None, "no_pref_reason": "invalid"}
     """
     annotations = {}
 
```

Comment:
I think we should assert here that the reference_preference is actually "text_a" or "text_b", otherwise this logic may break in unexpected (non-obvious hidden) ways